diff --git a/PM_phoenix-2.0/Defines.mk b/PM_phoenix-2.0/Defines.mk
index 6d2820a..272b1b1 100644
--- a/PM_phoenix-2.0/Defines.mk
+++ b/PM_phoenix-2.0/Defines.mk
@@ -96,7 +96,7 @@ LIB_DIR = lib
 INC_DIR = include
 TESTS_DIR = tests
 
-PMDKSRC = $(HOME)/../../../pmdk/src
+PMDKSRC = /SPP/pmdk/src
 
 ifneq ($(SPP_OFF),1)
 WRAP_LIST=-Wl,-wrap,free -Wl,-wrap,strcpy -Wl,-wrap,strcmp \
@@ -113,22 +113,17 @@ WRAP_LIST=-Wl,-wrap,free -Wl,-wrap,strcpy -Wl,-wrap,strcmp \
           -Wl,-wrap,pmemobj_memcpy_persist -Wl,-wrap,pmemobj_memmove \
           -Wl,-wrap,pmemobj_memset -Wl,-wrap,pmemobj_memset_persist
 
-SPPLIB=$(HOME)/../../../runtime
-SPPLIBSRC=${SPPLIB}/src
-SPPLIBOBJ=${SPPLIB}/obj
-LLVMROOT=$(HOME)/../../../llvm-project
-
 SPP_CFLAGS=-U_FORTIFY_SOURCE \
 -D_FORTIFY_SOURCE=0 \
 -Wno-array-bounds \
 -flto \
--Xclang -load -Xclang ${LLVMROOT}/build/lib/LLVMSPP.so \
+-Xclang -load -Xclang /llvm/lib/LLVMSPP.so \
 -DTAG_BITS=31 -fno-builtin
 
 SPP_LDFLAGS=-fuse-ld=gold \
 $(WRAP_LIST) \
-$(SPPLIBOBJ)/wrappers_spp.o \
--Xlinker ${SPPLIBOBJ}/spp.o
+/SPP/runtime/obj/wrappers_spp.o \
+-Xlinker /SPP/runtime/obj/spp.o
 else
 SPP_CFLAGS=
 SPP_LDFLAGS=
